# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/agnostic/bytestreams/H264.h
# Commit: b51b4dcd9a8a
# Full Hash: b51b4dcd9a8acdbd1474c6312a6ef07c54ed6531
# Author: Alastor Wu <alwu@mozilla.com>
# Date: 2019-09-04 04:20:48
# Regressor Bug: 1539182
# File Overlap Count: 2
# Description:
#   Bug 1539182 - check slice type to decide frame type. r=jya
#   
#   Sometime we would encouter a file putting all its samples on the type1 NAL, so we have to check its slice type in order to know if it's a keyframe or not.
#   
#   The slice type is defined in the ITU-T Rec H.264 table 7.3, and it could be found in the slice_header in table 7.3.3.
# ==============================================================================

diff -r e4deba1aa285 -r b51b4dcd9a8a dom/media/platforms/agnostic/bytestreams/H264.h
--- a/dom/media/platforms/agnostic/bytestreams/H264.h	Tue Sep 03 18:08:47 2019 +0000
+++ b/dom/media/platforms/agnostic/bytestreams/H264.h	Fri Aug 30 01:08:09 2019 +0000
@@ -36,6 +36,15 @@
   H264_NAL_SLICE_EXT_DVC = 21,
 };
 
+// According to ITU-T Rec H.264 (2017/04) Table 7.6.
+enum SLICE_TYPES {
+  P_SLICE = 0,
+  B_SLICE = 1,
+  I_SLICE = 2,
+  SP_SLICE = 3,
+  SI_SLICE = 4,
+};
+
 struct SPSData {
   bool operator==(const SPSData& aOther) const;
   bool operator!=(const SPSData& aOther) const;
